.clang-format

```diff
@@ -2,6 +2,10 @@ BasedOnStyle: Google
 ColumnLimit: 120
 IndentWidth: 4
 
+DerivePointerAlignment: false
```

Comment:
Hmmm.. I don't like changing these repo wide clang parameters as part of other PRs, but would like it to be a separate PR which also updates all the affected files so we can see the changes. Can you move it out of this PR?
Also, please comment on ALL options so that they're clear on what they do. For instance, I'm currently not sure on DerivePointerAlignment and what does AllowShortFunctionsOnASingleLine exactly do.

Reply:
I'll introduce .clang-format changes separately.